### PR TITLE
Update to core 1.5.0

### PR DESCRIPTION
**Pull request by tgoyne**
@bdash

### CODE REVIEW

File: tests/results.cpp [hunk 1]
```diff
@@ -320,7 +320,7 @@ TEST_CASE("results: notifications") {
     }
 
     // Sort in descending order
-    results = results.sort({{0}, {false}});
+    results = results.sort({*table, {{0}}, {false}});
```
Comment (bdash) [inline tests/results.cpp]: It's a little unfortunate that this change requires callers of `sort` to restate the table, when both `Results` and `List` already know it.

File: src/results.cpp [hunk 2]
```diff
@@ -484,9 +481,8 @@ TableView Results::get_tableview()
     REALM_UNREACHABLE();
 }
 
-Results Results::sort(realm::SortOrder&& sort) const
+Results Results::sort(realm::SortDescriptor&& sort) const
 {
-    REALM_ASSERT(sort.column_indices.size() == sort.ascending.size());
```
Comment (bdash) [inline src/results.cpp]: Is it worth asserting that the sort descriptor's table is correct here?

**Comment by bdash** [comment]
Unfortunate that `SortDescriptor` is a little less ergonomic for the bindings to use, but it's nice to not have to reinvent it in the object store.

👍

File: tests/results.cpp (hunk 1 above)
Comment (tgoyne) [inline tests/results.cpp]: It's definitely gross in the tests, but obj-c was using the table when constructing SortDescriptors anyway, so a convenience wrapper wouldn't help it. Might be worth adding something for the other bindings, though.

File: src/results.cpp (hunk 2 above)
Comment (tgoyne) [inline src/results.cpp]: That should probably be done *somewhere*, but the SortDescriptor currently doesn't actually know what table it's for. Maybe I can make it check on use at least.

**Comment by JadenGeller** [comment]
If I'm understanding correctly, constructor of `SortDescriptor` is gross because you have to state the `Table` even if the table is evident by context. Maybe it's worth introducing a `TransientSortDescriptor` (that stores column indices) that is taken as an argument by member functions of types associated with a table.

**Comment by JadenGeller** [comment]
Something like this is what I had in mind. I'm not sure it's an API improvement though. Duplicate types is kind of gross. https://github.com/realm/realm-object-store/commit/c02591e5f5d9da71a035bd917a9f71c0d4ef21d8

**Comment by JadenGeller** [comment]
At that point, it almost makes sense to skip introducing a new type and just make `sort` take the column indices and ascending vectors as arguments.

**Pull request merged**